src/main/java/org/broadinstitute/hellbender/utils/hdf5/HDF5PoNCreator.java

```diff
@@ -0,0 +1,712 @@
+            final double[] columnCounts = counts.getColumn(i);
+            columnMedians[i] = medianCalculator.evaluate(columnCounts);
+        }
+
+        // Calculate thresholds:
+        final double bottomExtremeThreshold = new Percentile(extremeColumnMedianCountPercentileThreshold).evaluate(columnMedians);
+        final double topExtremeThreshold = new Percentile(100 - extremeColumnMedianCountPercentileThreshold).evaluate(columnMedians);
+
+        // Determine kept and dropped column sets.
+        final Set<String> columnsToKeep = new HashSet<>(readCounts.columnNames().size());
+        final Map<String, Double> columnsToDrop = new LinkedHashMap<>((int) (extremeColumnMedianCountPercentileThreshold * 4.0 / 100.0));
+        for (int i = 0; i < columnMedians.length; i++) {
+            if (columnMedians[i] >= bottomExtremeThreshold && columnMedians[i] <= topExtremeThreshold) {
+                columnsToKeep.add(columnNames.get(i));
+            } else {
+                columnsToDrop.put(columnNames.get(i), columnMedians[i]);
+            }
+        }
+
+        // log and drop columns if it applies
+        if (columnsToKeep.isEmpty()) {
+            throw new UserException.BadInput("No column count left after applying the extreme counts outlier filter");
+        } else if (columnsToKeep.size() == columnNames.size()) {
+            logger.info(String.format("No column dropped due to extreme counts outside [%.10f, %.10f]", bottomExtremeThreshold, topExtremeThreshold));
+            return readCounts;
+        } else {
+            logger.info(String.format("Some columns dropped, %d of %d, as they are classified as having extreme median counts across targets outside [%.10f, %.10f]: e.g. %s",
+                    columnNames.size() - columnsToKeep.size(), columnNames.size(), bottomExtremeThreshold, topExtremeThreshold,
+                    columnsToDrop.entrySet().stream().limit(10).map(kv -> kv.getKey() + " (" + kv.getValue() + ")").collect(Collectors.joining(", "))));
+            return readCounts.subsetColumns(columnsToKeep);
+        }
+
+    }
+
+    /**
+     * Remove targets that have too many counts equal to 0.
+     * <p>
+     *     It will return a copy of the input read-count collection with such targets dropped.
+     * </p>
+     *
+     * @param readCounts the input read counts.
+     * @param maximumTargetZeros maximum number of counts equal to 0. per target tolerated.
+     * @return never {@code null}. It might be a reference to the input read-counts if there is
+     *   is no target to be dropped.
+     */
+    @VisibleForTesting
+    static ReadCountCollection removeTargetsWithTooManyZeros(final ReadCountCollection readCounts, final int maximumTargetZeros, final Logger logger) {
+        final Set<Target> targetsToKeep = new HashSet<>(readCounts.targets().size());
+        final RealMatrix counts = readCounts.counts();
+        for (int i = 0; i < counts.getRowDimension(); i++) {
```

Comment:
Hard using a predicate.  Just created a simple method and put that into a ``.filter(...)``
Done.